src/webots/gui/WbSimulationView.cpp

```diff
@@ -607,23 +614,23 @@ void WbSimulationView::makeMovie() {
 
   // reset current simulation mode
   WbSimulationState::instance()->setMode(currentMode);
-  updateFastModeOverlay();
+  updateBlackRenderingOverlay();
 }
 
-void WbSimulationView::switchToRunModeIfNecessary() {
+void WbSimulationView::showRenderingIfNecessary() {
   // remove "Fast Mode" overlay if necessary
-  if (WbSimulationState::instance()->isFast()) {
-    WbSimulationState::instance()->setMode(WbSimulationState::RUN);
-    mView3D->hideFastModeOverlay();
-    mNeedRestoreFastMode = true;
+  if (!WbSimulationState::instance()->isRendering()) {
+    WbSimulationState::instance()->setRendering(true);
+    mView3D->hideBlackRenderingOverlay();
+    mNeedToHideBlackRenderingOverlay = true;
```

Comment:
This variable name doesn't seem correct.
```suggestion
    mNeedToRestoreBlackRenderingOverlay = true;
```